### ISSUE TITLE
nixos/ddclient service broken in unstable

### ISSUE THREAD

**Issue by womfoo** [NixOS/nixpkgs#53035]
## Issue description

ddclient no longer starts after #48844 was merged cc @Infinisil @c0bw3b

```
Dec 29 23:05:46 silverspark ddclient[15004]: stat() on closed filehandle FD at /nix/store/b88bxc048az5hr8ndhvk4a9b728k5hl5-perl5.28.0-ddclient-3.8.3/bin/ddclient line 1067.
Dec 29 23:05:46 silverspark ddclient[15004]: Use of uninitialized value $mode in bitwise and (&) at /nix/store/b88bxc048az5hr8ndhvk4a9b728k5hl5-perl5.28.0-ddclient-3.8.3/bin/ddclient line 1068.
Dec 29 23:05:46 silverspark ddclient[15004]: readline() on closed filehandle FD at /nix/store/b88bxc048az5hr8ndhvk4a9b728k5hl5-perl5.28.0-ddclient-3.8.3/bin/ddclient line 1080.
Dec 29 23:05:46 silverspark ddclient[15004]: stat() on closed filehandle FD at /nix/store/b88bxc048az5hr8ndhvk4a9b728k5hl5-perl5.28.0-ddclient-3.8.3/bin/ddclient line 1067.
Dec 29 23:05:46 silverspark ddclient[15004]: Use of uninitialized value $mode in bitwise and (&) at /nix/store/b88bxc048az5hr8ndhvk4a9b728k5hl5-perl5.28.0-ddclient-3.8.3/bin/ddclient line 1068.
Dec 29 23:05:46 silverspark ddclient[15004]: readline() on closed filehandle FD at /nix/store/b88bxc048az5hr8ndhvk4a9b728k5hl5-perl5.28.0-ddclient-3.8.3/bin/ddclient line 1080.
Dec 29 23:05:46 silverspark ddclient[15004]: WARNING:  file /run/ddclient/ddclient.conf: Cannot open file '/run/ddclient/ddclient.conf'. (Permission denied)
Dec 29 23:05:46 silverspark systemd[1]: ddclient.service: Main process exited, code=exited, status=1/FAILURE
Dec 29 23:05:46 silverspark systemd[1]: ddclient.service: Failed with result 'exit-code'.
Dec 29 23:05:46 silverspark systemd[1]: Failed to start Dynamic DNS Client.
```

The service uses `DynamicUser = true` and chmodding the ddclient configuration file to `660` only makes it accessible to root which led to this new issue. 

I've commented DynamicUser on my local nixpkgs as a workaround for now but at the cost of running the service as root (we used to have dedicated uid/gids for ddclient) .

I've also tried adding `--owner=ddclient` to the `ExecStartPre` but the user doesn't seem to be available when it executes. Does anyone know of a way to set the file ownership with `DynamicUser`?

### Steps to reproduce

enable ddclient service on nixos-unstable

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.4, NixOS, 19.03.git.fe551d1 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(kranium): `"nixos"`
 - channels(root): `"nixos-18.09pre140682.5da85431fb1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
